src/app/organizations/organization-details/organization-details.component.ts

```diff
@@ -35,56 +33,42 @@ export class OrganizationDetailsComponent implements OnInit {
 
     ngOnInit() {
         const orgID = this.activeRoute.snapshot.params.id;
-        this.organizationService.getSingleOrganization(orgID).subscribe(org => {
-            this.organization = org;
-        }, err => {
-            displayBackendError(err);
-        });
-
-        this.walletService.getOrganizationWallet(orgID).subscribe(res => {
-            this.orgWallet = res;
-        }, err => {
-            if (err.status === 404) {
-                this.orgWallet = null;
-            } else if (err.error.err_code === '0851') {
-                swal('', 'The organization is being created. This can take up to a minute. Please check again later.', 'info')
-                    .then(() => {
-                        window.history.back();
-                    });
-            } else {
-                displayBackendError(err);
-            }
-        });
+        this.organization$ = this.organizationService.getSingleOrganization(orgID).pipe(this.handleError);
+        this.orgWallet$ = this.walletService.getOrganizationWallet(orgID).pipe(
+            catchError(err => {
+                if (err.status === 404) {
+                    return of(undefined);
+                } else if (err.error.err_code === '0851') {
+                    swal('', 'The organization is being created. This can take up to a minute. Please check again later.', 'info')
+                        .then(() => {
+                            window.history.back();
+                        });
+                } else {
+                    displayBackendError(err);
+                }
+                return EMPTY;
+            }));
 
-        this.organizationService.getMembersForOrganization(orgID)
-            .subscribe(res => {
-                this.orgMembers = res.members;
-            }, err => {
-                displayBackendError(err);
-            });
+        this.orgMember$ = this.refreshOrganizationMemberSubject.pipe(
+            switchMap(_ => this.organizationService.getMembersForOrganization(orgID)),
+            map(res => res.members)).pipe(this.handleError);
     }
 
-    inviteClicked() {
+    inviteClicked(orgUUID: string) {
         SpinnerUtil.showSpinner();
         const email = $('#email-invite-input').val();
-        this.organizationService.inviteUser(this.organization.uuid, email).subscribe(res => {
-            SpinnerUtil.hideSpinner();
-            swal('Success', 'Successfully invited user to organization', 'success');
-        }, err => {
-            SpinnerUtil.hideSpinner();
-            displayBackendError(err);
-        });
+        this.organizationService.inviteUser(orgUUID, email)
+            .subscribe(() => {
+                SpinnerUtil.hideSpinner();
```

Comment:
Hide spinner in finalize block. Additionally, it would be better that:
1. spinner is shown,
2. user is invited (successful request),
3. popup with success message is shown,
4. on OK, the spinner is hidden.

To make it like this, you can do:

```
this.organizationService.inviteUser(orgUUID, email).pipe(
    this.handleErrors(),
    switchMap(() => from(swal('Success', 'Successfully invited user to organization', 'success'))),
    finalize(() => SpinnerUtil.hideSpinner())
).subscribe()
```

It's more cleaner.
Consider that `swal` is a `Promise`, so we can wrap it in `from` operator to transform it to `Observable`.